### PR TITLE
feat: Generated unit tests for PercentDiscount and Discount abstract classes

### PR DESCRIPTION
**Pull request by marleomac3**

### CODE REVIEW

File: app/src/test/java/q4/team1/DiscountTest.java [hunk 1]
```diff
@@ -1,5 +1,43 @@
...
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+
+import q4.team1.exceptions.ItemNotFoundException;
+
 public class DiscountTest {
+    private ArrayList<CartItem> items;
+
+    private static class TestDiscount extends Discount {
```
Comment (Josephcabs) [inline app/src/test/java/q4/team1/DiscountTest.java:16]: Good use of abstraction in your previous files, it is displayed well here.

File: app/src/test/java/q4/team1/PercentDiscountTest.java [hunk 2]
```diff
@@ -1,5 +1,26 @@
...
+       items.add(item2);
+       PercentDiscount percentDiscount = new PercentDiscount(items, 0.5);
+
+       // Act
+       double discountAmount = percentDiscount.calculateDiscount();
 
+       // Assert
+        assertEquals(15.0, discountAmount);
+    }
 }
```
Comment (Josephcabs) [inline app/src/test/java/q4/team1/PercentDiscountTest.java:26]: Would add more tests, maybe an exception one for a negative discount and one to get the discount rates so we can cover all grounds and assume well rounded functionality

File: app/src/test/java/q4/team1/DiscountTest.java (hunk 1 above)
Comment (marleomac3) [reply app/src/test/java/q4/team1/DiscountTest.java:16]: thank you kind sir

File: app/src/test/java/q4/team1/PercentDiscountTest.java (hunk 2 above)
Comment (marleomac3) [reply app/src/test/java/q4/team1/PercentDiscountTest.java:52]: Just added the extra tests

**Review by Josephcabs — APPROVED** [review]
good work LJ
